script.js

```diff
@@ -0,0 +1,248 @@
+    let startSelected = this.textArea.selectionStart;
+    let endSelected = this.textArea.selectionEnd;
+    if (startSelected == endSelected) return false;
+    this.textArea.value = this.textArea.value.slice(0, startSelected) + this.textArea.value.slice(endSelected);
+    this.textArea.selectionEnd = this.textArea.selectionStart = startSelected;
+    return true;
+  }
+
+  backspaceSymbol() {
+    if (this.removeSelectedPiece()) return;
+    let cursorPosition = this.textArea.selectionStart;
+    if (!cursorPosition) return;
+    this.textArea.value = this.textArea.value.slice(0, cursorPosition-1) + this.textArea.value.slice(cursorPosition);
+    this.textArea.selectionEnd = this.textArea.selectionStart = cursorPosition-1;
+  }
+
+  deleteSymbol() {
+    if (this.removeSelectedPiece()) return;
+    let cursorPosition = this.textArea.selectionStart;
+    this.textArea.value = this.textArea.value.slice(0, cursorPosition) + this.textArea.value.slice(cursorPosition+1);
+    this.textArea.selectionEnd = this.textArea.selectionStart = cursorPosition;
+  }
+
+  pressKeyDown(key) {
+    switch (key.id) {
+      case 'ControlLeft':
+      case 'MetaLeft':
+      case 'AltLeft':
+      case 'AltRight':
+      case 'ControlRight':
+        key.classList.add('active-key');
+        break;
+      case 'Delete':
+        key.classList.add('active-key');
+        this.deleteSymbol();
+        break;
+      case 'Backspace':
+        key.classList.add('active-key');
+        this.backspaceSymbol();
+        break;
+      case 'ShiftRight':
+      case 'ShiftLeft':
+        if(event.repeat) return;
+        key.classList.add('active-key');
+        this.isShift = true;
+        this.changeLayout();
+        this.keyboard.classList.toggle('upper-case');
+        this.isCaseUp = !this.isCaseUp;
+        break;
+      case 'CapsLock':
+        if(event.repeat) return;
+        key.classList.toggle('active-key');
+        this.keyboard.classList.toggle('upper-case');
+        this.isCaseUp = !this.isCaseUp;
+        break;
+      default:
+        key.classList.add('active-key');
+        this.insertSymbol(key);
+    }
+  }
+
+  pressKeyUp(key) {
+    switch (key.id) {
+      case 'ShiftRight':
+      case 'ShiftLeft':
+        key.classList.remove('active-key');
+        this.isShift = false;
+        this.changeLayout();
+        this.keyboard.classList.toggle('upper-case');
+        this.isCaseUp = !this.isCaseUp;
+        break;
+      case 'CapsLock':
+        break;
+      default:
+        key.classList.remove('active-key');
+    }
+  }
+};
+
+window.addEventListener('DOMContentLoaded', () => {
+
+  let divWrapper = document.createElement('div');
+  divWrapper.classList.add('wrapper');
+
+  let textArea = document.createElement('textarea');
+  textArea.classList.add('textarea');
+  divWrapper.append(textArea);
+  document.body.append(divWrapper);
+
+  virtualKeyboard = new VirtualKeyboard();
+
+  virtualKeyboard.divWrapper = divWrapper;
```

Comment:
I think it is better to pass this through constructor

Reply:
I have corrected